cmd/assaf-http/main.go

```diff
@@ -310,53 +263,54 @@ func subscribe(queueName string, sessions chan chan session, consumeRequests cha
 
 		log.Printf("subscribed...")
 
-		curr := 0
-		seen := 0
-		for msg := range deliveries {
```

Comment:
This is the heart of the design change, instead of looping over all the requests in the incoming `consumeRequests` channel (which is lock free approach) we are now getting them with their explicit key from a map. although the lock free approach is tempting it brought with a it a bunch of state checks that didn't work well in a lock free manner.